### ISSUE TITLE
Setting keypath 'data' on none-isolated Component corrupts model

### ISSUE THREAD

**Issue by LeJared** [ractivejs/ractive#3030]
## Description:
Executing this in a component having `isolated: false`
```JS
this.set({
    a:'a',
    data: {
        b:'b',
        c:'c',
    }
});
```
results in this model beeing actually set (calling `this.get();`):
```JS
{ a: "a", c: "c", b: "b" }
```
When `isolated` is set to `true` the correct model gets set:
```JS
{ a: "a", data: {b: "b",  c: "c"} }
```

## Versions affected:
0.9.2

## Platforms affected:
Firefox 54, Chrome 59

## Reproduction:
https://jsfiddle.net/lejared/ntjkeLzd/

**Comment by evs-chris** [comment]
Implicit mappings will try to resolve using the usual template reference resolution logic, so you can either block that at the `set` by using `set({ ... }, { isolated: true })` or turn off [member resolution](https://ractive.js.org/api/#resolveinstancemembers) for the component or globally.

**Comment by LeJared** [comment]
I think this bug can be closed as it seems to be fixed in 0.9.3
